File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

bool isnum(string num);
int main(int argc, string argv[])
{
    if (argc == 2 && isnum(argv[1]))
    {
        int key = (atoi(argv[1])) % 26;
        string pt = get_string("plaintext:  ");
        printf("ciphertext: ");
        int i = 0;
        while (pt[i] != 0)
        {
            if (islower(pt[i]))
            {
                printf("%c", 'a' + (pt[i] - 'a' + key) % 26);
            }    
            else if (isupper(pt[i]))
            {
                printf("%c", 'A' + (pt[i] - 'A' + key) % 26);
            }
            else
            {
                printf("%c", pt[i]);
            }
            i++;
        }
        printf("\n");
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}
bool isnum(string num)
{
    for (int i = 0; i < strlen(num); i++)
    {
        if (!isdigit(num[i]))
        {
            return 0;
        }
    }
    return 1;
}

File: caesar.py
from cs50 import get_string
from sys import argv, exit

if len(argv) != 2 or argv[1].isdigit() == False:
    exit("Usage: ./caesar key")

key = int(argv[1]) % 26
pt = get_string("plaintext:  ")
print("ciphertext: ", end="")

for i in pt:
    if (i.islower() == True):
        print(chr(ord('a') + (ord(i) - ord('a') + key) % 26), end="")
    elif (i.isupper() == True):
        print(chr(ord('A') + (ord(i) - ord('A') + key) % 26), end="")
    else:
        print(i, end="")
print()